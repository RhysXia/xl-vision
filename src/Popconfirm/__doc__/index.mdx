# Popconfirm

点击元素，弹出气泡式的确认框。

## 何时使用

目标元素的操作需要用户进一步的确认时，在目标元素附近弹出浮层提示，询问用户。

和`confirm`弹出的全屏居中模态对话框相比，交互形式更轻量。

:::demo 基本用法

基本用法

```jsx
import React from 'react'
import { Popconfirm, Button } from 'xl-vision'

export default () => {
  const message = 'Are you sure?'

  return (
    <div className='container'>
      <div className='top'>
        <Popconfirm placement='top-start' message={message}>
          <Button theme='primary'>tl</Button>
        </Popconfirm>
        <Popconfirm placement='top' message={message}>
          <Button theme='primary'>top</Button>
        </Popconfirm>
        <Popconfirm placement='top-end' message={message}>
          <Button theme='primary'>tr</Button>
        </Popconfirm>
      </div>
      <div className='center'>
        <div className='left'>
          <Popconfirm placement='left-start' message={message}>
            <Button theme='primary'>lt</Button>
          </Popconfirm>
          <Popconfirm placement='left' message={message}>
            <Button theme='primary'>left</Button>
          </Popconfirm>
          <Popconfirm placement='left-end' message={message}>
            <Button theme='primary'>lb</Button>
          </Popconfirm>
        </div>
        <div className='right'>
          <Popconfirm placement='right-start' message={message}>
            <Button theme='primary'>rt</Button>
          </Popconfirm>
          <Popconfirm placement='right' message={message}>
            <Button theme='primary'>right</Button>
          </Popconfirm>
          <Popconfirm placement='right-end' message={message}>
            <Button theme='primary'>rb</Button>
          </Popconfirm>
        </div>
      </div>
      <div className='bottom'>
        <Popconfirm placement='bottom-start' message={message}>
          <Button theme='primary'>bl</Button>
        </Popconfirm>
        <Popconfirm placement='bottom' message={message}>
          <Button theme='primary'>bottom</Button>
        </Popconfirm>
        <Popconfirm placement='bottom-end' message={message}>
          <Button theme='primary'>br</Button>
        </Popconfirm>
      </div>
    </div>
  )
}
```

```scss
.container {
  &::deep {
    .xl-button {
      margin: 10px;
      min-width: 100px;
    }
  }

  .top,
  .bottom {
    margin-left: 100px + 10px;
  }
  .center {
    display: flex;
  }
  .left,
  .right {
    display: flex;
    flex-direction: column;
  }
  .right {
    margin-left: (100px + 10px) * 3;
  }
}
```

:::

:::demo 不同的触发方式

通过`trigger`指定触发方式，默认值是`hover`，如果有复杂需求，可以设置为`custom`，以获得完全的控制能力。

```jsx
import React from 'react'
import { Popconfirm, Button } from 'xl-vision'
export default () => {
  const message = 'Are you sure?'
  const [visible, setVisible] = React.useState(false)

  const onClick = React.useCallback(() => {
    setVisible((prev) => !prev)
  }, [])

  return (
    <Button.Group theme='info'>
      <Popconfirm placement='top' trigger='hover' message={message}>
        <Button>hover</Button>
      </Popconfirm>
      <Popconfirm placement='top' trigger='click' message={message}>
        <Button>click</Button>
      </Popconfirm>
      <Popconfirm placement='top' trigger='focus' message={message}>
        <Button>focus</Button>
      </Popconfirm>
      <Popconfirm placement='top' trigger='contextMenu' message={message}>
        <Button>contextMenu</Button>
      </Popconfirm>
      <Popconfirm placement='top' trigger='custom' visible={visible} message={message}>
        <Button onClick={onClick}>custom(double click)</Button>
      </Popconfirm>
    </Button.Group>
  )
}
```

:::

:::demo 延迟触发

通过`hideDelay`和`showDelay`指定延时触发时间。

```jsx
import React from 'react'
import { Popconfirm, Button } from 'xl-vision'
export default () => {
  const message = 'Are you sure?'

  return (
    <Button.Group theme='secondary'>
      <Popconfirm placement='top' showDelay={2000} message={message}>
        <Button>show after 2s</Button>
      </Popconfirm>
      <Popconfirm placement='top' hideDelay={2000} message={message}>
        <Button>hide after 2s</Button>
      </Popconfirm>
    </Button.Group>
  )
}
```

:::

:::demo 禁用

禁止`popconfirm`展示。

```jsx
import React from 'react'
import { Popconfirm, Button } from 'xl-vision'
export default () => {
  const message = 'Are you sure?'

  return (
    <Popconfirm disabled message={message}>
      <Button theme='primary'>button</Button>
    </Popconfirm>
  )
}
```

:::

## Popconfirm API

> 更多参数请参考<Link to="Popper">**Popper**</Link>属性

| 参数 | 说明 | 类型 | 可选值 | 默认值 | 是否必填 |
| --- | --- | --- | --- | --- | --- |
| clsPrefix | 自定义类名前缀 | string | - | - | false |
| message | 内容 | React.ReactNode | - | - | true |
| footer | 自定义底部按钮 | React.ReactNode | - | - | false |
| confirmText | 确认按钮文字 | string | - | - | false |
| cancelText | 取消按钮文字 | string | - | - | false |
| confirmButtonProps | 确认按钮属性 | <Link to="Button#Button API">ButtonProps</Link> | - | - | false |
| confirmButtonProps | 取消按钮属性 | <Link to="Button#Button API">ButtonProps</Link> | - | - | false |
| onConfirm | 点击确认事件 | (e)=>void | - | - | false |
| onCancel | 点击取消事件 | (e)=>void | - | - | false |
| placement | 指定弹出框的方位 | `auto`&#x007c;`auto-start`&#x007c;`auto-end`&#x007c;`top`&#x007c;`top-start`&#x007c;`top-end`&#x007c;`right`&#x007c;`right-start`&#x007c;`right-end`&#x007c;`bottom`&#x007c;`bottom-start`&#x007c;`bottom-end`&#x007c;`left`&#x007c;`left-start`&#x007c;`left-end` | - | `auto` | false |
| transitionClasses | 动画 class，参看<Link to='CSSTransition'>CSSTransition</Link> | boolean | - | `xl-popconfirm--zoom` | false |
| getPopupContainer | 指定弹出框的父容器 | string &#x007c; HTMLElement &#x007c; (() => string &#x007c; HTMLElement) | - | 默认为`body` | false |
| visible | 是否显示弹出框 | boolean | - | false | false |
| onVisibleChange | 弹出框状态改变时的回调 | (visible: boolean) => void | - | - | false |
| hideDelay | 延时关闭 | number | - | - | false |
| showDelay | 延时显示 | number | - | - | false |
| trigger | 触发方式 | `hover`&#x007c;`focus`&#x007c;`click`&#x007c;`contextMenu`&#x007c;`custom` | - | `click` | false |
| disablePopupEnter | 禁止鼠标进入弹出框 | boolean | - | - | false |
| forceRender | 强制弹出框立即挂载到 dom 中，即使弹出框还没有显示 | boolean | - | - | false |
| offset | 弹出框偏移 | `number` | - | `10` | false |
| style | 弹出框样式 | React.CSSProperties | - | - | false |
| className | 弹出框类名 | string | - | - | false |
| flip | 反转弹出框以避免遮挡 | boolean | HTMLElement | HTMLElement[] | (() =>)boolean | HTMLElement | HTMLElement[] | - | true | false |
| preventOverflow | 调整弹出框以避免遮挡 | boolean | HTMLElement | HTMLElement[] | (() =>)boolean | HTMLElement | HTMLElement[] | - | true | false |
| disableGpuAcceleration | 禁用 gpu 渲染 | boolean | - | - | false |
| disableTransformOrigin | 禁用默认的`transform-origin`行为 | boolean | - | - | false |
| disabled | 禁用 | boolean | - | - | false |
