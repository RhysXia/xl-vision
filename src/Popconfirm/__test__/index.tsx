import { mount, ReactWrapper } from 'enzyme'
import * as React from 'react'
import Popconfirm from '..'
import { act } from 'react-dom/test-utils'
import wait from '../../../test/wait'

describe('Popconfirm', () => {
  let wrapper: ReactWrapper

  beforeAll(() => {
    jest.useRealTimers()
  })

  afterEach(() => {
    wrapper.unmount()
  })

  it('render', () => {
    wrapper = mount(
      <Popconfirm className='popup' visible={true} message='message'>
        <button className='btn'>button</button>
      </Popconfirm>
    )

    expect(wrapper.render()).toMatchSnapshot()
  })

  it('测试onConfirm', async () => {
    const fn = jest.fn()
    wrapper = mount(
      <Popconfirm className='popup' visible={true} message='message' onConfirm={fn}>
        <button className='btn'>button</button>
      </Popconfirm>
    )

    await act(() => wait(50))

    const popup = document.querySelector('.popup') as HTMLElement

    expect(popup.style.display).toBe('')

    wrapper.find('button.xl-button--theme-primary').simulate('click')

    await act(() => wait(50))

    expect(fn.mock.calls.length).toBe(1)
    expect(popup.style.display).toBe('none')
  })

  it('测试onCancel', async () => {
    const fn = jest.fn()
    wrapper = mount(
      <Popconfirm className='popup' visible={true} message='message' onCancel={fn}>
        <button className='btn'>button</button>
      </Popconfirm>
    )

    await act(() => wait(50))

    const popup = document.querySelector('.popup') as HTMLElement

    expect(popup.style.display).toBe('')

    wrapper.find('button.xl-button--theme-default').simulate('click')

    await act(() => wait(50))

    expect(fn.mock.calls.length).toBe(1)
    expect(popup.style.display).toBe('none')
  })
})
