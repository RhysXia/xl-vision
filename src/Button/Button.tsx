import classnames from 'classnames'
import * as PropTypes from 'prop-types'
import React from 'react'
import { Reload, IconProps } from '../icon'
import ButtonContext from './ButtonContext'
import ConfigContext from '../ConfigProvider/ConfigContext'
import Ripple from '../Ripple'

export type ButtonSize = 'large' | 'default' | 'small'

export interface BaseButtonProps {
  children: React.ReactNode
  className?: string
  dashed?: boolean
  disabled?: boolean
  ghost?: boolean
  href?: string
  htmlType?: 'submit' | 'reset' | 'button'
  loading?: boolean
  long?: boolean
  plain?: boolean
  clsPrefix?: string
  shape?: 'circle' | 'round'
  size?: ButtonSize
  type?: 'default' | 'primary' | 'success' | 'warning' | 'error' | 'text'
  prefixIcon?: React.ReactElement<IconProps>
  suffixIcon?: React.ReactElement<IconProps>
}

export type ButtonProps = BaseButtonProps &
  Omit<React.AnchorHTMLAttributes<HTMLAnchorElement>, 'type'> &
  Omit<React.ButtonHTMLAttributes<HTMLButtonElement>, 'type'>

const Button: React.FunctionComponent<ButtonProps> = React.forwardRef<
  HTMLButtonElement & HTMLAnchorElement,
  ButtonProps
>((props, ref) => {
  const { clsPrefix: rootClsPrefix } = React.useContext(ConfigContext)

  const {
    dashed,
    disabled,
    ghost,
    plain,
    href,
    htmlType,
    loading,
    long,
    shape,
    type = 'default',
    className,
    children,
    clsPrefix = `${rootClsPrefix}-button`,
    onClick,
    prefixIcon,
    suffixIcon,
    ...others
  } = props

  let { size } = React.useContext(ButtonContext)

  // 强制使用ButtonGroup的size
  size = size || others.size || 'default'

  delete others.size

  const classes = classnames(
    {
      [clsPrefix]: true,
      [`${clsPrefix}--${type}`]: true,
      [`${clsPrefix}--${shape}`]: shape,
      [`${clsPrefix}--size-${size}`]: size,
      [`${clsPrefix}--dashed`]: dashed,
      [`${clsPrefix}--plain`]: plain,
      [`${clsPrefix}--ghost`]: ghost,
      [`${clsPrefix}--long`]: long,
      [`${clsPrefix}--disabled`]: disabled,
      [`${clsPrefix}--loading`]: loading
    },
    className
  )

  const onClickHandler = React.useCallback(
    (e: React.MouseEvent<HTMLButtonElement & HTMLAnchorElement, MouseEvent>) => {
      // 在disabled和loading状态下是不允许触发点击事件的
      if (disabled || loading) {
        e.preventDefault()
        return
      }
      onClick && onClick(e)
    },
    [onClick, disabled, loading]
  )

  const prefixNode = React.useMemo(() => {
    const classes = classnames(`${clsPrefix}__icon`, `${clsPrefix}__icon--prefix`)
    if (loading) {
      return <Reload className={classes} spin={true} />
    }
    if (!prefixIcon) {
      return
    }
    const node = React.cloneElement(prefixIcon, {
      className: classnames(prefixIcon.props.className, classes)
    })
    return node
  }, [loading, prefixIcon, clsPrefix])

  const suffixNode = React.useMemo(() => {
    if (!suffixIcon) {
      return
    }
    const classes = classnames(`${clsPrefix}__icon`, `${clsPrefix}__icon--suffix`)
    const node = React.cloneElement(suffixIcon, {
      className: classnames(suffixIcon.props.className, classes)
    })
    return node
  }, [suffixIcon, clsPrefix])

  const childrenWrapper = (
    <>
      {prefixNode}
      {formatChildren(children)}
      {suffixNode}
      <Ripple />
    </>
  )

  const allProps = {
    ...others,
    children: childrenWrapper,
    className: classes,
    disabled,
    href,
    type: htmlType,
    onClick: onClickHandler
  }

  if (href) {
    return <a ref={ref} {...allProps} />
  }
  return <button ref={ref} {...allProps} />
})

Button.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  dashed: PropTypes.bool,
  disabled: PropTypes.bool,
  ghost: PropTypes.bool,
  href: PropTypes.string,
  htmlType: PropTypes.oneOf(['submit', 'reset', 'button']),
  loading: PropTypes.bool,
  long: PropTypes.bool,
  plain: PropTypes.bool,
  clsPrefix: PropTypes.string,
  shape: PropTypes.oneOf<'circle' | 'round'>(['circle', 'round']),
  size: PropTypes.oneOf(['large', 'default', 'small']),
  type: PropTypes.oneOf(['default', 'primary', 'success', 'warning', 'error', 'text']),
  prefixIcon: PropTypes.element,
  suffixIcon: PropTypes.element
}

export default Button

const formatChildren = (children: React.ReactNode) => {
  return React.Children.map(children, (child) => {
    if (typeof child === 'string') {
      return <span>{child}</span>
    }
    return child
  })
}
