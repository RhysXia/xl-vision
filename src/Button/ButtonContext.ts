import React from 'react'
import { ButtonTheme, ButtonVariant } from './Button'

export type ButtonContextProps = {
  theme?: ButtonTheme
  variant?: ButtonVariant
  disableElevation?: boolean
  disableRipple?: boolean
  groupClsPrefix?: string
  size?: string
}

export default React.createContext<ButtonContextProps>({})
