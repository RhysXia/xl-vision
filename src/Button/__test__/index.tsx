import React from 'react'
import { mount } from 'enzyme'
import Button from '..'
import { Close } from '../../icon'

describe('Button', () => {
  it('基本测试', () => {
    const wrapper = mount(
      <div>
        <Button>button</Button>
        <Button theme='error'>button</Button>
        <Button variant='outlined'>button</Button>
        <Button prefixIcon={<Close />}>button</Button>
        <Button suffixIcon={<Close />}>button</Button>
        <Button prefixIcon={<Close />}></Button>
        <Button suffixIcon={<Close />}></Button>
        <Button long={true}>button</Button>
        <Button loading={true}>button</Button>
        <Button loading={true} prefixIcon={<Close />}></Button>
        <Button loading={true} prefixIcon={<Close />}>
          button
        </Button>
      </div>
    )

    expect(wrapper.render()).toMatchSnapshot()
  })
})

describe('ButtonGroup', () => {
  it('基本测试', () => {
    const wrapper = mount(
      <Button.Group theme='primary' variant='contained'>
        <Button>button</Button>
        <Button theme='error'>button</Button>
        <Button variant='outlined'>button</Button>
        <Button prefixIcon={<Close />}>button</Button>
        <Button suffixIcon={<Close />}>button</Button>
        <Button prefixIcon={<Close />}></Button>
        <Button suffixIcon={<Close />}></Button>
        <Button loading={true}>button</Button>
        <Button loading={true} prefixIcon={<Close />}></Button>
        <Button loading={true} prefixIcon={<Close />}>
          button
        </Button>
      </Button.Group>
    )

    expect(wrapper.render()).toMatchSnapshot()
  })
})
