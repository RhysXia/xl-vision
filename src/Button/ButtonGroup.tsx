import classnames from 'classnames'
import PropTypes from 'prop-types'
import React from 'react'
import { ButtonProps } from './Button'
import ConfigContext from '../ConfigProvider/ConfigContext'
import ButtonContext, { ButtonContextProps } from './ButtonContext'

export interface ButtonGroupProps
  extends Omit<ButtonContextProps, 'groupClsPrefix'>,
    React.HTMLAttributes<HTMLDivElement> {
  children: React.ReactElement<ButtonProps> | Array<React.ReactElement<ButtonProps>>
  clsPrefix?: string
  round?: boolean
}

const ButtonGroup: React.FunctionComponent<ButtonGroupProps> = (props) => {
  const { clsPrefix: rootClsPrefix } = React.useContext(ConfigContext)

  const {
    round,
    className,
    clsPrefix = `${rootClsPrefix}-button-group`,
    disableRipple,
    disableElevation,
    theme,
    variant,
    size,
    ...others
  } = props
  const classes = classnames(
    {
      [clsPrefix]: true,
      [`${clsPrefix}--round`]: round
    },
    className
  )

  return (
    <ButtonContext.Provider
      value={{
        groupClsPrefix: `${clsPrefix}`,
        size,
        disableRipple,
        disableElevation,
        theme,
        variant
      }}
    >
      <div {...others} className={classes} />
    </ButtonContext.Provider>
  )
}

ButtonGroup.displayName = 'ButtonGroup'

ButtonGroup.propTypes = {
  theme: PropTypes.oneOf([
    'default',
    'primary',
    'error',
    'warning',
    'secondary',
    'success',
    'info'
  ]),
  variant: PropTypes.oneOf(['contained', 'text', 'outlined']),
  size: PropTypes.oneOf(['small', 'medium', 'large']),
  disableRipple: PropTypes.bool,
  disableElevation: PropTypes.bool,
  className: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.element.isRequired,
    PropTypes.arrayOf(PropTypes.element.isRequired)
  ]).isRequired,
  clsPrefix: PropTypes.string,
  round: PropTypes.bool
}

export default ButtonGroup
