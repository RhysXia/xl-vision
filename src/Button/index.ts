import Button from './Button'
import ButtonGroup from './ButtonGroup'

export { ButtonProps, ButtonTheme, ButtonVariant } from './Button'
export { ButtonGroupProps } from './ButtonGroup'

const ButtonWithGroup = Button as typeof Button & {
  Group: typeof ButtonGroup
}

ButtonWithGroup.Group = ButtonGroup

export default ButtonWithGroup
