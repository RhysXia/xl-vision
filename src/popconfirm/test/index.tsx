import { mount } from 'enzyme'
import * as React from 'react'
import { act } from 'react-testing-library'
import Popconfirm from '..'
import Button from '../../button'

describe('Popconfirm', () => {
  it('Show Popconfirm', () => {
    const Demo = ({ visible }: { visible: boolean }) => (
      <Popconfirm visible={visible} content={'content'}>
        <Button>Button</Button>
      </Popconfirm>
    )

    expect.assertions(2)

    const wrapper = mount(<Demo visible={false}/>)

    expect(wrapper).toMatchSnapshot()

    wrapper.setProps({
      visible: true
    })

    act(() => {
      expect(wrapper).toMatchSnapshot()
    })
  })
})
