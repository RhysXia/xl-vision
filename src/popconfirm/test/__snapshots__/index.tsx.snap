// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popconfirm Show Popconfirm 1`] = `
<Demo
  visible={false}
>
  <xl-popconfirm
    content="content"
    visible={false}
  >
    <xl-tooltip
      allowPopupEnter={true}
      arrowSize={14}
      content={
        <React.Fragment>
          <div
            className="xl-popconfirm__message"
          >
            <xl-fas-exclamation-circle
              className="xl-popconfirm__icon"
            />
            <span
              className="xl-popconfirm__content"
            >
              content
            </span>
          </div>
          <div
            className="xl-popconfirm__action"
          >
            <ForwardRef
              className="xl-popconfirm__btn"
              onClick={[Function]}
              size="small"
              type="primary"
            >
              Yes
            </ForwardRef>
            <ForwardRef
              className="xl-popconfirm__btn"
              onClick={[Function]}
              size="small"
              type="default"
            >
              No
            </ForwardRef>
          </div>
        </React.Fragment>
      }
      onVisibleChange={[Function]}
      prefixCls="xl-popconfirm"
      visible={false}
    >
      <xl-popper
        allowPopupEnter={true}
        arrow={[Function]}
        offset={5}
        onVisibleChange={[Function]}
        overlayStyle={[Function]}
        popup={[Function]}
        transitionName="xl-popconfirm--fade"
        visible={false}
      >
        <xl-button
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <button
            className="xl-button xl-button--default xl-button--size-default"
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <span
              key=".0"
            >
              Button
            </span>
          </button>
        </xl-button>
        <Portal
          containerInfo={
            <body>
              <div
                style="left: 0px; position: absolute; top: 0px; padding-top: 5px;"
              />
            </body>
          }
        >
          <div
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "left": 0,
                "paddingTop": 5,
                "position": "absolute",
                "top": 0,
              }
            }
          >
            <xl-css-transition
              beforeAppear={[Function]}
              beforeEnter={[Function]}
              classNames="xl-popconfirm--fade"
              isAppear={true}
              show={false}
            >
              <xl-transition
                afterAppear={[Function]}
                afterEnter={[Function]}
                afterLeave={[Function]}
                appear={[Function]}
                appearCancelled={[Function]}
                beforeAppear={[Function]}
                beforeEnter={[Function]}
                beforeLeave={[Function]}
                enter={[Function]}
                enterCancelled={[Function]}
                isAppear={true}
                leave={[Function]}
                leaveCancelled={[Function]}
                show={false}
              />
            </xl-css-transition>
          </div>
        </Portal>
      </xl-popper>
    </xl-tooltip>
  </xl-popconfirm>
</Demo>
`;

exports[`Popconfirm Show Popconfirm 2`] = `
<Demo
  visible={true}
>
  <xl-popconfirm
    content="content"
    visible={true}
  >
    <xl-tooltip
      allowPopupEnter={true}
      arrowSize={14}
      content={
        <React.Fragment>
          <div
            className="xl-popconfirm__message"
          >
            <xl-fas-exclamation-circle
              className="xl-popconfirm__icon"
            />
            <span
              className="xl-popconfirm__content"
            >
              content
            </span>
          </div>
          <div
            className="xl-popconfirm__action"
          >
            <ForwardRef
              className="xl-popconfirm__btn"
              onClick={[Function]}
              size="small"
              type="primary"
            >
              Yes
            </ForwardRef>
            <ForwardRef
              className="xl-popconfirm__btn"
              onClick={[Function]}
              size="small"
              type="default"
            >
              No
            </ForwardRef>
          </div>
        </React.Fragment>
      }
      onVisibleChange={[Function]}
      prefixCls="xl-popconfirm"
      visible={false}
    >
      <xl-popper
        allowPopupEnter={true}
        arrow={[Function]}
        offset={5}
        onVisibleChange={[Function]}
        overlayStyle={[Function]}
        popup={[Function]}
        transitionName="xl-popconfirm--fade"
        visible={false}
      >
        <xl-button
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <button
            className="xl-button xl-button--default xl-button--size-default"
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <span
              key=".0"
            >
              Button
            </span>
          </button>
        </xl-button>
        <Portal
          containerInfo={
            <body>
              <div
                style="left: 0px; position: absolute; top: 0px; padding-top: 5px;"
              />
            </body>
          }
        >
          <div
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "left": 0,
                "paddingTop": 5,
                "position": "absolute",
                "top": 0,
              }
            }
          >
            <xl-css-transition
              beforeAppear={[Function]}
              beforeEnter={[Function]}
              classNames="xl-popconfirm--fade"
              isAppear={true}
              show={false}
            >
              <xl-transition
                afterAppear={[Function]}
                afterEnter={[Function]}
                afterLeave={[Function]}
                appear={[Function]}
                appearCancelled={[Function]}
                beforeAppear={[Function]}
                beforeEnter={[Function]}
                beforeLeave={[Function]}
                enter={[Function]}
                enterCancelled={[Function]}
                isAppear={true}
                leave={[Function]}
                leaveCancelled={[Function]}
                show={false}
              />
            </xl-css-transition>
          </div>
        </Portal>
      </xl-popper>
    </xl-tooltip>
  </xl-popconfirm>
</Demo>
`;
